src/main/java/com/syntifi/crypto/key/Secp256k1PublicKey.java

```diff
@@ -114,32 +114,20 @@ public Boolean verify(byte[] message, byte[] signature) {
 
     /**
      * Gets a short key
+     * There's around a 50% chance the elliptical curve algo will generate a 65 byte
+     * public key instead of 66 byte.
+     * Luckily the algo pads the first byte as zero when this happens
+     * startBit determines this
      *
      * @param key the key as a byte array
      * @return short key as byte array
      */
     public static byte[] getShortKey(final byte[] key) {
         final BigInteger pubKey = new BigInteger(key);
         final String pubKeyPrefix = pubKey.testBit(0) ? "03" : "02";
-        final byte[] pubKeyBytes = Arrays.copyOfRange(key, 0, (AlgorithmTag.SECP256K1.getLength() - 1));
+        final int startBit = key[0] == (byte) 0 ? 1 : 0;
+        final byte[] pubKeyBytes = Arrays.copyOfRange(key, startBit, (AlgorithmTag.SECP256K1.getLength() - 1) + startBit);
         return Hex.decode(pubKeyPrefix + Hex.encode(pubKeyBytes));
```

Comment:
Isn't this a bit inefficient. Would be better to use a buffer or array population you dont need to encode then decode.